lib/spack/spack/test/cmd/load.py

```diff
@@ -17,61 +18,85 @@
 install = SpackCommand("install")
 location = SpackCommand("location")
 
-pytestmark = pytest.mark.not_on_windows("does not run on windows")
-
 
 def test_manpath_trailing_colon(
     install_mockery, mock_fetch, mock_archive, mock_packages, working_env
 ):
+    (shell, set_command, commandsep) = (
+        ("--bat", 'set "%s=%s"', "\n")
+        if sys.platform == "win32"
+        else ("--sh", "export %s=%s", ";\n")
+    )
+
     """Test that the commands generated by load add the MANPATH prefix
     inspections. Also test that Spack correctly preserves the default/existing
     manpath search path via a trailing colon"""
     install("mpileaks")
 
-    sh_out = load("--sh", "mpileaks")
-    lines = sh_out.split("\n")
-    assert any(re.match(r"export MANPATH=.*:;", ln) for ln in lines)
+    sh_out = load(shell, "mpileaks")
+    lines = sh_out.split(commandsep)
+    assert any(re.match(set_command % ("MANPATH", ".*" + os.pathsep), ln) for ln in lines)
+    os.environ["MANPATH"] = "/tmp/man" + os.pathsep
 
-    os.environ["MANPATH"] = "/tmp/man:"
-
-    sh_out = load("--sh", "mpileaks")
-    lines = sh_out.split("\n")
-    assert any(re.match(r"export MANPATH=.*:/tmp/man:;", ln) for ln in lines)
+    sh_out = load(shell, "mpileaks")
+    lines = sh_out.split(commandsep)
+    assert any(
+        re.match(set_command % ("MANPATH", ".*" + os.pathsep + "/tmp/man" + os.pathsep), ln)
+        for ln in lines
+    )
 
 
-def test_load_recursive(install_mockery, mock_fetch, mock_archive, mock_packages, working_env):
+@pytest.mark.parametrize(
+    "shell,set_command_prefix,set_command_postfix,commandsep",
+    (
+        [("--bat", 'set "%s=', '%s"', "\n")]
```

Comment:
This decomposition is hard to follow IMO, in particular the splitting into `set_command_prefix` and `set_command_postfix` (this makes it slightly more confusing than the other parameterizations): is it possible to rewrite the test in terms of a single `set_command`?